### PR TITLE
Add missing exception to Model.fit() docstring

### LINKED ISSUE #46492: documentation missing, tf.keras.Model.fit shuffle argument is being ignored when passing a tf.data.dataset
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit

## Description of issue (what needs changing):

There's an issue with the shuffle argument description, it doesn't state that the "shuffle" argument is being ignored
when the argument "x" is a tf.data.dataset in tf.keras.model.fit.

### Clear description

I was inspecting the source code in tf.keras.model.fit and found that when the input arg x to the [data handler](https://github.com/tensorflow/tensorflow/blob/v2.4.0/tensorflow/python/keras/engine/training.py#L1050) is a tf.data.dataset, it ends up using the [dataset adapter](https://github.com/tensorflow/tensorflow/blob/582c8d236cb079023657287c318ff26adb239002/tensorflow/python/keras/engine/data_adapter.py#L671) which silently ignores the shuffle argument. 

This issue is not clearly explained in the description of the shuffle argument below:

_"""Boolean (whether to shuffle the training data before each epoch) or str (for 'batch'). This argument is ignored when x is a generator. 'batch' is a special option for dealing with the limitations of HDF5 data; it shuffles in batch-sized chunks. Has no effect when steps_per_epoch is not None."""_

I would recommend to write something like this instead "_This argument is ignored when x is a generator or a tf.data.Dataset._" 

I can contribute by fixing the docs if you agree that this should be done.

Thank you in advance for your review!

### PR DESCRIPTION
**Pull request by Suraj-Upadhyay**
The shuffle argument to the fit() method in the Model class gets ignored
if the input x to the method is a tf.data.Dataset object. But this isn't
documented in the docstring of the method.

Signed-off-by: Suraj Upadhyay <usuraj35@gmail.com>

Closes #46492

### CODE REVIEW

**Review by rchao — APPROVED** [review]
Thanks for the contribution!
